### PR TITLE
add userCohort option for engagement banner tests

### PR DESCRIPTION
**Pull request by jranks123**
## What does this change?
We want the ability to create a banner test which will only display to existing supporters. This PR adds this functionality.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js [hunk 1]
```diff
@@ -206,11 +206,18 @@ const canShow = (): Promise<boolean> => {
             getVisitCount() >= params.minArticlesBeforeShowingBanner;
         const userAlreadyGivesUsMoney = userIsSupporter();
         const bannerIsBlockedForEditorialReasons = pageShouldHideReaderRevenue();
+        const showBecauseUserAlreadyGivesUsMoney =
```
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: This is how the epic does it these days. Personally I find it clearer:

```javascript
    const isCompatibleUser = test.onlyShowToExistingSupporters
        ? userIsSupporter()
        : !userIsSupporter();
```

Comment (jranks123) [reply static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: 👍 good shout, I've reused the function now

Comment (jranks123) [reply static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: (Note: what would you say to `onlyShowToExistingSupporters === userIsSupporter();`? Terser but arguably less clear, interested in your opinion)

**Review by tomrf1 — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js [hunk 2]
```diff
@@ -126,19 +126,18 @@ const isCompatibleWithLiveBlogEpic = (page: Object): boolean =>
 const pageShouldHideReaderRevenue = () =>
     config.get('page.shouldHideReaderRevenue');
 
+const isCompatibleUser = (onlyShowToExistingSupporters: boolean) =>
+    onlyShowToExistingSupporters ? userIsSupporter() : !userIsSupporter();
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js]: `onlyShowToExistingSupporters` suggests that if it's false it shows to everyone, perhaps the name could be improved? Not sure if an enum is lightweight enough to be usable here as boolean is prone to error

**Comment by johnduffell** [comment]
looks great to me, just a small naming comment 👍

**Review by johnduffell — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js (hunk 2 above)
Comment (joelochlann) [reply static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js]: Yeah me & Tom have been round the houses with this one a lot and never found a nice way of doing it with a boolean. So I actually really like the idea of something like

`type UserCohort = 'OnlyExistingSupporters' | 'OnlyNonSupporters' | 'Everyone';`

Comment (joelochlann) [reply static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js]: or `TargetedUserCohort`
or `TargetsUsers`

Comment (joelochlann) [reply static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js]: I guess to do that properly we would set it that way in the test types (epic & banner) too. And then maybe eventually in the sheet, though we could leave that for the moment

Comment (jranks123) [reply static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js]: I ❤️ it. Have a look and see what you think @johnduffell @joelochlann

**Review by joelochlann — APPROVED** [review]
❤️

**Review by tomrf1 — APPROVED** [review]
much nicer naming!

**Pull request merged**